src/admin/mod.rs

```diff
@@ -337,6 +341,57 @@ async fn handle_metrics(reg: Arc<Mutex<Registry>>, _req: Request<Body>) -> Respo
         .unwrap()
 }
 
+async fn handle_loglevel(req: Request<Body>) -> Response<Body> {
+    match *req.method() {
+        hyper::Method::GET => {
+            let loglevel = match telemetry::get_current() {
+                Ok(level) => format!("current log level is {}\n", level),
+                Err(e) => format!("failed to get log level, err: {}\n", e),
+            };
+            Response::builder()
+                .status(hyper::StatusCode::OK)
+                .header(hyper::header::CONTENT_TYPE, "text/plain")
+                .body(loglevel.into())
+                .unwrap()
+        }
+        hyper::Method::POST => match get_new_loglevel(req).await {
+            Ok(level) => match telemetry::set_level(level) {
+                Ok(return_str) => Response::builder()
+                    .status(hyper::StatusCode::OK)
+                    .header(hyper::header::CONTENT_TYPE, "text/plain")
+                    .body(return_str.into())
+                    .unwrap(),
+                Err(return_str) => Response::builder()
+                    .status(hyper::StatusCode::METHOD_NOT_ALLOWED)
```

Comment:
METHOD_NOT_ALLOWED doesn't seem quite right

Reply:
changed to INTERNAL_SERVER_ERROR

Reply:
Doesn't seem changed? Should probably be BAD_REQUEST though